opennav_following/src/following_server.cpp

```diff
@@ -0,0 +1,676 @@
+// Copyright (c) 2024 Open Navigation LLC
+// Copyright (c) 2024 Alberto J. Tudela Roldán
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "angles/angles.h"
+#include "opennav_docking_core/docking_exceptions.hpp"
+#include "opennav_following/following_server.hpp"
+#include "nav2_util/geometry_utils.hpp"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
+#include "tf2/utils.hpp"
+
+using namespace std::chrono_literals;
+using rcl_interfaces::msg::ParameterType;
+using std::placeholders::_1;
+
+namespace opennav_following
+{
+
+FollowingServer::FollowingServer(const rclcpp::NodeOptions & options)
+: nav2::LifecycleNode("following_server", "", options)
+{
+  RCLCPP_INFO(get_logger(), "Creating %s", get_name());
+
+  declare_parameter("controller_frequency", 50.0);
```

Comment:
Once we approve here, this is something we'll want a docs PR for in the configuration guide + migration guide to bring this to folks' mind. If you have a nice video / gif of the following (I think you already showed some) that would be a great place to use it in the migration & configuration guides. Maybe a tutorial for how to use and important features like the pose vs TF, static obstacle timeout, rotate vs wait for target when lost, etc?